elasticdl_preprocessing/layers/to_ragged.py

```diff
@@ -0,0 +1,58 @@
+import tensorflow as tf
+
+_COMMA_SEP = ","
+
+
+class ToRagged(tf.keras.layers.Layer):
+    """Converts a `Tensor` or `RaggedTensor` to a `RaggedTensor`,
+    dropping ignore_value cells. If the input's dtype is string, split
+    the string elements to convert the input to `RaggedTensor` firstly.
+
+    Example (Integer):
+    ```python
+        layer = ToRagged()
+        inp = tf.constant([[1], [-1], [4]], tf.int64)
+        out = layer(inp)
+        [[1], [], [4]]
+    ```
+
+    Example (String):
+    ```python
+        ayer = ToRagged()
+        inp = tf.constant([["1,2,3"], ["4,5"], [""]])
+        out = layer(inp)
+    ```
+    The expected output is `[["1", "2", "3"], ["4", "5"], []]`
+
+    Arguments:
+        sep: Valid if the input's dtype is string.
+        ignore_value: Entries in inputs equal to this value will be
+            absent from the output `RaggedTensor`. If `None`, default value of
+            input's dtype will be used ('' for `str`, -1 for `int`).
+
+    Input shape: A numeric or string `Tensor` or `RaggedTensor` of shape
+        `[batch_size, d1, ..., dm]`
+
+    Output shape: An `RaggedTensor` with the same shape as inputs
+    """
+
+    def __init__(self, sep=_COMMA_SEP, ignore_value=None):
+        super(ToRagged, self).__init__()
+        self.sep = sep
+        self.ignore_value = ignore_value
+
+    def call(self, inputs):
```

Comment:
What will happen if inputs is a SparseTensor?

Reply:
It may raise an unknown error. We need to check the input type and raise an error is the type is not `Tensor` or `RaggedTensor`.